File: app/src/main/java/com/esorokin/lantector/presentation/view/history/HistoryView.java
package com.esorokin.lantector.presentation.view.history;

import java.util.List;

import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy;
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType;
import com.esorokin.lantector.model.data.DetectedLanguageText;
import com.esorokin.lantector.presentation.error.UserError;
import com.esorokin.lantector.presentation.view.BaseView;

@StateStrategyType(AddToEndSingleStrategy.class)
public interface HistoryView extends BaseView {
	void showError(UserError error);

	void hideError();

	void showItems(List<DetectedLanguageText> items);

	void hideItems();
}

File: app/src/main/java/com/esorokin/lantector/presentation/view/history/HistoryView.kt
package com.esorokin.lantector.presentation.view.history

import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType
import com.esorokin.lantector.model.data.DetectedLanguageText
import com.esorokin.lantector.presentation.error.UserError
import com.esorokin.lantector.presentation.view.BaseView

@StateStrategyType(AddToEndSingleStrategy::class)
interface HistoryView : BaseView {
    fun showError(error: UserError)
    fun hideError()

    fun showItems(items: List<DetectedLanguageText>)
    fun hideItems()
}